pixeltable/functions/twelvelabs.py

```diff
@@ -69,20 +64,125 @@ async def embed(
         Add a computed column `embed` for an embedding of a string column `input`:
 
         >>> tbl.add_computed_column(
-        ...     embed=embed(model_name='Marengo-retrieval-2.7', text=tbl.input)
+        ...     embed=embed(model_name='marengo3.0', text=tbl.input)
         ... )
     """
+    env.Env.get().require_package('twelvelabs')
+    import twelvelabs
+
     cl = _twelvelabs_client()
-    res = await cl.embed.create(
-        model_name=model_name, text=text, text_truncate=text_truncate, audio_file=audio, **kwargs
+    res: twelvelabs.EmbeddingSuccessResponse
+    if image is None:
+        # Text-only
+        res = await cl.embed.v_2.create(
+            input_type='text', model_name=model_name, text=twelvelabs.TextInputRequest(input_text=text)
+        )
+    else:
+        b64str = to_base64(image, format=('png' if image.has_transparency_data else 'jpeg'))
+        res = await cl.embed.v_2.create(
+            input_type='text_image',
+            model_name=model_name,
+            text_image=twelvelabs.TextImageInputRequest(
+                media_source=twelvelabs.MediaSource(base_64_string=b64str), input_text=text
+            ),
+        )
+    if not res.data:
+        raise pxt.Error(f"Didn't receive embedding for text: {text}\n{res}")
+    vector = res.data[0].embedding
+    return np.array(vector, dtype='float32')
+
+
+@embed.overload
+async def _(model_name: str, *, image: pxt.Image) -> pxt.Array[np.float32] | None:
+    env.Env.get().require_package('twelvelabs')
+    import twelvelabs
+
+    cl = _twelvelabs_client()
+    b64_str = to_base64(image, format=('png' if image.has_transparency_data else 'jpeg'))
+    res = await cl.embed.v_2.create(
+        input_type='image',
+        model_name=model_name,
+        image=twelvelabs.ImageInputRequest(media_source=twelvelabs.MediaSource(base_64_string=b64_str)),
     )
-    if text is not None:
-        if res.text_embedding is None:
-            raise pxt.Error(f"Didn't receive embedding for text: {text}")
-        vector = res.text_embedding.segments[0].float_
-        return np.array(vector, dtype=np.float64)
-    # TODO: handle audio and image, once we know how to get a non-error response
-    return None
+    if not res.data:
+        raise pxt.Error(f"Didn't receive embedding for image: {image}\n{res}")
+    vector = res.data[0].embedding
+    return np.array(vector, dtype='float32')
+
+
+@embed.overload
+async def _(
+    model_name: str,
+    *,
+    audio: pxt.Audio,
+    start_sec: float | None = None,
+    end_sec: float | None = None,
+    embedding_option: list[Literal['audio', 'transcription']] | None = None,
+    embedding_scope: list[Literal['clip', 'asset']] | None = None,
```

Comment:
I don't think this parameter makes sense, since we don't expose the segmentation aspect (and I don't think we should, our Audio-/VideoSplitter are more useful for that). We should simply pass 'asset'.